Controllers/UserController.cs

```diff
@@ -68,15 +73,39 @@ public async Task<IActionResult> Signup(RegisterDTO model)
 
             var user = new Employee
             {
+                PersonalCode = model.PersonalCode,
+                FirstName = model.FirstName,
+                LastName = model.LastName,
                 Username = model.Username,
                 Email = model.Email,
+                Password = model.Password,
                 Department = model.RoleId,
-                FirstName = "Test",
-                LastName = "Testing",
-                PersonalCode = "39000000000",
-                Status = EmployeeStatusId.Employed
+                BirthDate = model.BirthDate,
+                Status = EmployeeStatusId.Employed,
+                RegisterDate = DateTime.Now
             };
 
+            switch (model.RoleId)
+            {
+                case DepartmentId.Pharmacy:
+                    user.Pharmacy = Context.Pharmacy.Where(z => z.Id == model.PharmacyWarehouseOrTruck).FirstOrDefault();
+                    break;
+                case DepartmentId.Warehouse:
+                    user.Warehouse = Context.Warehouse.Where(z => z.Id == model.PharmacyWarehouseOrTruck).FirstOrDefault();
```

Comment:
Maybe we could simplify this LINQ as `Context.Warehouse.FirstOrDefault(z => z.Id == model.PharmacyWarehouseOrTruck);`?